### PR TITLE
API: Allow to change orders

### PR DESCRIPTION
**Pull request by raphaelm**
- [x] REST-y operations on positions
  - [x] PATCH
  - [x] POST
  - [x] DELETE
  - [x] regenerate secret
- [x] Transactional API
  - [x] comprehensive tests
  - [x] docs

### CODE REVIEW

File: src/pretix/api/serializers/orderchange.py [hunk 1]
```diff
@@ -0,0 +1,423 @@
...
+            return
+        self.fields['order'].queryset = self.context['event'].orders.all()
+        self.fields['item'].queryset = self.context['event'].items.all()
+        self.fields['subevent'].queryset = self.context['event'].subevents.all()
+        self.fields['seat'].queryset = self.context['event'].seats.all()
+        self.fields['variation'].queryset = ItemVariation.objects.filter(item__event=self.context['event'])
+        if 'order' in self.context:
+            del self.fields['order']
+
+    def validate(self, data):
```
Comment (wiffbi) [inline src/pretix/api/serializers/orderchange.py:70]: Is this missing a `super().validate(data)`?

**Review by wiffbi — COMMENTED** [review]
I have to admit this is a rather complex PR. I took some time to try to understand the flow of the API-requests, serializers, etc. and mainly concentrated on the tests to understand, how it works. I actually just stumbled upon one little thing regarding validation, but I am not sure whether that is actually a problem. I think everything should work as intended, but then again I did not/could not follow everything through deeply enough. Sorry!

File: src/pretix/api/serializers/orderchange.py (hunk 1 above)
Comment (raphaelm) [reply src/pretix/api/serializers/orderchange.py:70]: Hmm, I think that the parent function doesn't do anything, but it is more correct to still call it, will change it.

**Comment by crackwitz** [comment]
The docs aren't clear on this: is the API capable of creating a position and also create addons on that position, in the same request?

**Pull request merged**